core/src/main/java/io/kestra/core/secret/SecretService.java

```diff
@@ -3,15 +3,26 @@
 import io.kestra.core.exceptions.IllegalVariableEvaluationException;
 import jakarta.inject.Singleton;
 
+import javax.annotation.PostConstruct;
 import java.util.Base64;
+import java.util.Map;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Singleton
 public class SecretService {
+    private Map<String, String> decodedSecrets;
+
+    @PostConstruct
+    private void postConstruct() {
+        decodedSecrets = System.getenv().entrySet().stream().filter(entry -> entry.getKey().startsWith("SECRETS_")).collect(Collectors.toMap(
+            entry -> entry.getKey().substring(8).toUpperCase(),
```

Comment:
Should be toLowerCase I think, and you can use the length of the prefix constant instead of hardcoding 8 ;)

Reply:
I get them as uppercase because env var will probably be as uppercase when set up. Whatever I enforce both the map input and the get as uppercase for now, is it ok ?